specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2024-06-02-preview/kubernetesResources.json

```diff
@@ -0,0 +1,70 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ContainerServiceClient",
+    "description": "The Container Service Client.",
+    "version": "2024-06-02-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "KubernetesResource": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties for the kubernetes object.",
+          "$ref": "#/definitions/KubernetesObject"
+        }
+      },
+      "description": "Kubernetes Resource."
+    },
+    "KubernetesObject": {
+      "type": "object",
+      "properties": {
+        "k8sKind": {
+          "type": "string",
+          "description": "namespace, pod, node, etc."
+        },
+        "properties": {
```

Comment:
This `properties` property is a `map[string]string`. Is that correct? Are the maps always flat, or is there nesting?

If there's nesting, I think this needs to be just `type: object`, without the `additionalProperties`